### PR TITLE
libwebsockets: Update to 3.1.0

### PR DESCRIPTION
**Pull request by neheb**
Remove upstreamed patch and backport another one.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

Maintainer: @karlp 
Compile tested: ar71xx

Temporarily removed 020-fix-travis.patch to see if it can be fixed properly.

### CODE REVIEW

File: libs/libwebsockets/Makefile [hunk 1]
```diff
@@ -8,21 +8,19 @@
...
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
 PKG_SOURCE_URL:=https://codeload.github.com/warmcat/libwebsockets/tar.gz/v$(PKG_VERSION)?
-PKG_HASH:=cb0cdd8d0954fcfd97a689077568f286cdbb44111883e0a85d29860449c47cbf
-
-PKG_SOURCE_VERSION:=v$(PKG_VERSION)
-
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(BUILD_VARIANT)/$(PKG_SOURCE_SUBDIR)
+PKG_HASH:=db948be74c78fc13f1f1a55e76707d7baae3a1c8f62b625f639e8f2736298324
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(BUILD_VARIANT)/$(PKG_NAME)-$(PKG_VERSION)
```
Comment (karlp) [inline libs/libwebsockets/Makefile]: was there any reason to change this either?  It just makes noise really, even if it's "one less variable" ?

**Comment by neheb** [comment]
It's more consistent with other packages

**Comment by hnyman** [comment]
@karlp updated it to 3.1.0 via 05ec863154

**Comment by karlp** [comment]
I'd forgotten this pr existed, I get notifications from sdtowa anyway.  I only handled the update.  The ssl ecc things are separate, and were not merged here.

**Comment by neheb** [comment]
Not going to bother respinning this. The ECC patch will be part of the next release anyway.

**Pull request closed without merging**